lib/Dialect/StableHLO/Utils/ShardyUtils.cpp

```diff
@@ -850,7 +850,10 @@ convertCustomCallToShardingConstraint(mlir::ModuleOp &rootModule,
     mlir::stablehlo::CustomCallOp customCallOp =
         mlir::cast<mlir::stablehlo::CustomCallOp>(op);
     auto callTargetName = customCallOp.getCallTargetNameAttr();
-    if (callTargetName != gspmd_utils::kShardingCustomCallTargetName) {
+    // Handle both @Sharding (from xs.mark_sharding) and @tt.sharding_constraint
+    // (from custom ops)
+    if (callTargetName != gspmd_utils::kShardingCustomCallTargetName &&
+        callTargetName != kTTShardingConstraintTargetName) {
```

Comment:
:warning: **clang\-diagnostic\-error** :warning:
use of undeclared identifier `` kTTShardingConstraintTargetName ``; did you mean `` sharding_utils::kTTShardingConstraintTargetName ``?
```suggestion
        callTargetName != sharding_utils::kTTShardingConstraintTargetName) {
```